src/Adapter/MSTest.CoreAdapter/Helpers/ReflectHelper.cs

```diff
@@ -200,6 +200,20 @@ public override object InitializeLifetimeService()
             return null;
         }
 
+        internal static T[] GetAttributes<T>(MethodBase methodBase, bool inherit)
+            where T : Attribute
+        {
+            Attribute[] attributeArray = GetCustomAttributes(methodBase, typeof(T), inherit);
+
+            var attributes = attributeArray as T[];
```

Comment:
In `~/src/Adapter/PlatformServices.Desktop/Utilities/DesktopReflectionUtility.cs` it might return a `T[]`, and that might be the most common path for this method.